### PR TITLE
Document that client-go is not yet compatible with dep

### PR DESCRIPTION
**Pull request by caesarxuchao**
@sttts could you proof read this? Thank you.

### CODE REVIEW

**Comment by caesarxuchao** [comment]
cc @rickypai @danderson

File: INSTALL.md [hunk 1]
```diff
@@ -143,38 +143,11 @@ After modifying, run `glide up -v` again to re-populate your /vendor directory.
...
-one of the plugins (for example, the auth plugins). So you may need to run `dep
-ensure` again if you start importing a plugin that you weren't using before.
+### Dep (Not supported yet!)
+
+[dep](https://github.com/golang/dep) is an up-and-coming dependency management
+tool, which has the goal of being accepted as part of the standard go toolchain.
+However, client-go does not work well with `dep` yet. To support `dep`, we need
+to fix at least two issues:
+1. publish native `Gopkg.toml` in client-go and other k8s.io repos, like `k8s.io/apimachinery`;
+2. find a way to express transitive constraints (see https://github.com/golang/dep/issues/1124).
```
Comment (sttts) [inline INSTALL.md]: Looks good. I would add: If you use dep today, you have to specify client-go dependencies manually in Gopkg.toml.

The client-go vendoring reality looks like this: https://twitter.com/the_sttts/status/924931147003547648. So many people use dep (more or less successfully, more or less due to luck that the master branches of our dependencies happen to work with our code today).

Comment (caesarxuchao) [reply INSTALL.md]: Done.

File: INSTALL.md [hunk 2]
```diff
@@ -143,38 +143,15 @@ After modifying, run `glide up -v` again to re-populate your /vendor directory.
...
+
+To use `dep` with client-go safely today, you have to specify client-go
+dependencies manually in Gopkg.toml with the versions listed in
+[Godeps.json](./Godeps/Godeps.json).
```
Comment (cheftako) [inline INSTALL.md]: We should probably repeat that this is not yet supported. We should also set expectations as to the cost of migrating from the work around to the final solution.

File: INSTALL.md [hunk 3]
```diff
@@ -143,38 +143,20 @@ After modifying, run `glide up -v` again to re-populate your /vendor directory.
...
+[dep](https://github.com/golang/dep) is an up-and-coming dependency management
+tool, which has the goal of being accepted as part of the standard go toolchain.
+However, client-go does **NOT** work well with `dep` yet. To support `dep`, we
+need to fix at least two issues:
+1. publish native `Gopkg.toml` in client-go and other k8s.io repos, like `k8s.io/apimachinery`;
+2. find a way to express transitive constraints (see https://github.com/golang/dep/issues/1124).
+
+As a workaround, which may or may not be worthwhile, you can specify all
+client-go dependencies manually as
+[override](https://github.com/golang/dep/blob/master/docs/Gopkg.toml.md#override)
```
Comment (caesarxuchao) [inline INSTALL.md]: @sttts is this accurate?

Comment (sttts) [reply INSTALL.md]: yes

**Comment by caesarxuchao** [comment]
Thank you both for the review.

**Pull request merged**